### ISSUE TITLE
May-NULL plus may-zero yields non-NULL pointer

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#1762]
This is a minimization of an issue found by @Robotechnic:
```c
int main(void) {
    int *p;
    int i;
    int *q = p + i;
    *q = 42; // TODO WARN (may NULL dereference)
}
```
We currently do not output a NULL dereference warning from base, even though `p` may be `NULL` and `i` may be 0.
It looks like our pointer arithmetic misses this possibility, because we get that `q → {?}`, which means non-`NULL` only, but no `NULL`.
